bot/exts/moderation/defcon.py

```diff
@@ -124,134 +136,143 @@ async def defcon_group(self, ctx: Context) -> None:
+        self._update_notifier()
+
+        action = Action.DURATION_UPDATE
+
+        expiry_message = ""
+        if expiry:
+            expiry_message = f"for the next {humanize_delta(relativedelta(expiry, datetime.utcnow()))}"
+
+        await self.channel.send(
+            f"{action.value.emoji} DEFCON threshold updated; accounts must be "
+            f"{humanize_delta(self.threshold)} old to join the server {expiry_message}."
+        )
+        await self._send_defcon_log(action, author)
+        self._update_channel_topic()
+
+        self._log_threshold_stat(threshold)
+
+    async def _remove_threshold(self) -> None:
+        """Resets the threshold back to 0."""
+        await self._update_threshold(self.bot.user, relativedelta(days=0))
+
+    @staticmethod
+    def _stringify_relativedelta(delta: relativedelta) -> str:
+        """Convert a relativedelta object to a duration string."""
+        units = [("years", "y"), ("months", "m"), ("days", "d"), ("hours", "h"), ("minutes", "m"), ("seconds", "s")]
+        return "".join(f"{getattr(delta, unit)}{symbol}" for unit, symbol in units if getattr(delta, unit)) or "0s"
+
+    def _log_threshold_stat(self, threshold: relativedelta) -> None:
+        """Adds the threshold to the bot stats in days."""
+        threshold_days = relativedelta_to_timedelta(threshold).total_seconds() / SECONDS_IN_DAY
+        self.bot.stats.gauge("defcon.threshold", threshold_days)
+
+    async def _send_defcon_log(self, action: Action, actor: User) -> None:
         """Send log message for DEFCON action."""
         info = action.value
         log_msg: str = (
             f"**Staffer:** {actor.mention} {actor} (`{actor.id}`)\n"
-            f"{info.template.format(days=self.days.days)}"
+            f"{info.template.format(threshold=humanize_delta(self.threshold))}"
         )
         status_msg = f"DEFCON {action.name.lower()}"
 
-        if e:
-            log_msg += (
-                "**There was a problem updating the site** - This setting may be reverted when the bot restarts.\n\n"
-                f"```py\n{e}\n```"
-            )
-
         await self.mod_log.send_log_message(info.icon, info.color, status_msg, log_msg)
 
+    def _update_notifier(self) -> None:
+        """Start or stop the notifier according to the DEFCON status."""
+        if self.threshold != relativedelta(days=0) and self.expiry is None and not self.defcon_notifier.is_running():
+            log.info("DEFCON notifier started.")
+            self.defcon_notifier.start()
+
+        elif (self.threshold == relativedelta(days=0) or self.expiry is not None) and self.defcon_notifier.is_running():
+            log.info("DEFCON notifier stopped.")
+            self.defcon_notifier.cancel()
+
+    @tasks.loop(hours=1)
+    async def defcon_notifier(self) -> None:
+        """Routinely notify moderators that DEFCON is active."""
+        await self.channel.send(f"Defcon is on and is set to {humanize_delta(self.threshold)}.")
```

Comment:
Won't that run even if defcon is disabled?

Reply:
There isn't a concept of disabled anymore. If you mean when it's 0, then no, it will be disabled by _update_notifier

Reply:
Alright, good enough for me.